front/src/modules/simulationResult/components/SpaceTimeChart/index.tsx

```diff
@@ -51,42 +67,75 @@ const CHART_MIN_HEIGHT = 250;
 
 export type SpaceTimeChartProps = {
   allowancesSettings?: AllowancesSettings;
-  dispatchUpdateDepartureArrivalTimes?: DispatchUpdateDepartureArrivalTimes;
-  dispatchUpdateChart?: DispatchUpdateChart;
-  dispatchUpdateMustRedraw?: DispatchUpdateMustRedraw;
-  dispatchUpdateSelectedTrainId?: DispatchUpdateSelectedTrainId;
-  updateTimePosition?: DispatchUpdateTimePositionValues;
   initialHeight?: number;
   inputSelectedTrain?: Train;
   selectedProjection?: OsrdSimulationState['selectedProjection'];
   simulation?: SimulationSnapshot;
   simulationIsPlaying?: boolean;
-  onOffsetTimeByDragging?: (trains: Train[], offset: number, timePosition: Date) => void;
   onSetBaseHeight?: (newHeight: number) => void;
   isDisplayed?: boolean;
 };
 
 export default function SpaceTimeChart(props: SpaceTimeChartProps) {
   const ref = useRef<HTMLDivElement>(null);
   const rndContainerRef = useRef<Rnd>(null);
+  const dispatch = useDispatch();
+
+  const allowancesSettingsFromStore = useSelector(getAllowancesSettings);
+  const inputSelectedTrainFromStore = useSelector(getSelectedTrain);
+  // implement selector for all selected trains ids
+  const selectedProjectionFromStore = useSelector(getSelectedProjection);
+  const simulationFromStore = useSelector(getPresentSimulation);
```

Comment:
As both of this component parents are already calling these 2 selectors, should we pass them as props (like the hoc did) ? Or it goes against the goal of making this component independant ? @clarani 
```suggestion
  // implement selector for all selected trains ids
  const selectedProjectionFromStore = useSelector(getSelectedProjection);
```

Reply:
I would prefer that these props comes from the parent component, what do you think ?